### DISCUSSION TITLE
Set root path on config

### DISCUSSION THREAD

**Discussion by ben182** [PlatformCommunity/flysystem-bunnycdn#30] [Ideas]
Hi,

what would be a really helpful feature is if the adapter would respect the root config attribute that you can specify to put your files right into a specific directory, like the S3 adapter does.

**Comment by sifex** [comment]
Hi Ben, there's no `rootDirectory` variable for this package, I haven't quite needed it yet. 

Can you prove that you have a lot of app logic that requires that? I would probably have to put through a breaking change / 4th parameter in the `BunnyCDNClient` constructor with default's in order to make this work.

**Reply by ben182** [reply]
Hi Alex, thanks for this discussion.
In summary: I have a CMS and want to create different subdirectories on the same bunny storage where users have different view access. Of course, this could also be done via multiple storage, but then I would also have to connect multiple pull zones and the whole thing becomes unnecessarily complicated. It would therefore be very handy to separate the same storage with different root folders. Unfortunately, I have no way of solving this via manual prefixes in all adapter calls, as the CMS does the heavy lifting. Regardless, I think it would also be a good feature, as S3 also supports this, and we could be an excellent alternative if people want to migrate from S3. On the subject of breaking change: Couldn't this also be solved by simply setting the fourth parameter to optional? That way, existing installations would remain unaffected.

**Comment by sifex** [comment]
v3.2.0 Tagged and released, let me know if that resolves your issue. Many thanks to @tinect
